projects/table/src/lib/table/ngx-table.component.ts

```diff
@@ -523,10 +523,10 @@ export class NgxTableComponent
 					// Iben: Select all items when this control is selected, if not, unselect them all
 					if (selected) {
 						this.rowsFormGroup.patchValue(
-							this.data.reduce((previousValue, _, index) => {
+							this.data.reduce((previousValue, currentValue) => {
 								return {
 									...previousValue,
-									[`${index}`]: true,
+									[currentValue[this.selectableKey]]: true,
```

Comment:
De selectableKey is optioneel, waardoor dat als die niet wordt voorzien er wel nog altijd met de index gewerkt wordt. Je zal dus beide moeten kunnen ondersteunen; en dit best in een aparte util in de `handle-form-change.util` file.